instrumentation/metric/registry.go

```diff
@@ -85,39 +87,28 @@ func (r *inMemoryRegistry) Get(metricName string) metric {
 			return metric
 		}
 	}
-	panic(errors.Errorf("a metric with the name of %s is not registered", metricName))
+	return nil
```

Comment:
`Get` locks the map for a full scan. this is beyond the scope of this PR but consider opening an issue or a TODO comment to add a lookup map that will speed up these queries (you can write to the map in "register", and every time you get some item from that list check if it's still in `metrics` so that you never return items that were removed, while always returning the object that was most recently registered...

if you agree with this approach better defer to a different PR (post - v2)...

Reply:
the only get is in the status page. i can change to read lock in whatever next pr i do